File: sample/src/main/java/me/xiaopan/assemblyadaptersample/itemfactory/HeaderGroupItemFactory.java
package me.xiaopan.assemblyadaptersample.itemfactory;

import android.content.Context;
import android.view.ViewGroup;
import android.widget.TextView;

import me.xiaopan.assemblyadapter.AssemblyGroupItem;
import me.xiaopan.assemblyadapter.AssemblyGroupItemFactory;
import me.xiaopan.assemblyadaptersample.R;

public class HeaderGroupItemFactory extends AssemblyGroupItemFactory<HeaderGroupItemFactory.HeaderGroupItem> {
    @Override
    public boolean isTarget(Object data) {
        return data instanceof String;
    }

    @Override
    public HeaderGroupItem createAssemblyItem(ViewGroup parent) {
        return new HeaderGroupItem(R.layout.list_item_header, parent);
    }

    public class HeaderGroupItem extends AssemblyGroupItem<String> {
        private TextView textView;

        public HeaderGroupItem(int itemLayoutId, ViewGroup parent) {
            super(itemLayoutId, parent);
        }

        @Override
        protected void onFindViews() {
            textView = (TextView) getItemView();
        }

        @Override
        protected void onConfigViews(Context context) {

        }

        @Override
        protected void onSetData(int position, String s) {
            textView.setText(s);
        }
    }
}

File: sample/src/main/java/me/xiaopan/assemblyadaptersample/itemfactory/HeaderGroupItemFactory.kt
package me.xiaopan.assemblyadaptersample.itemfactory

import android.content.Context
import android.view.ViewGroup
import android.widget.TextView

import me.xiaopan.assemblyadapter.AssemblyGroupItem
import me.xiaopan.assemblyadapter.AssemblyGroupItemFactory
import me.xiaopan.assemblyadaptersample.R

class HeaderGroupItemFactory : AssemblyGroupItemFactory<HeaderGroupItemFactory.HeaderGroupItem>() {
    override fun isTarget(data: Any): Boolean {
        return data is String
    }

    override fun createAssemblyItem(parent: ViewGroup): HeaderGroupItem {
        return HeaderGroupItem(R.layout.list_item_header, parent)
    }

    inner class HeaderGroupItem(itemLayoutId: Int, parent: ViewGroup) : AssemblyGroupItem<String>(itemLayoutId, parent) {

        override fun onConfigViews(context: Context) {

        }

        override fun onSetData(position: Int, s: String) {
            (itemView as TextView).text = s
        }
    }
}
